tutorial/markdown/aspnet/semantic-kernel/semantic-kernel-tutorial.md

```diff
@@ -0,0 +1,388 @@
+glossary.DefinitionEmbedding = embedding.Vector;
+```
+
+For more details on embedding generation in Semantic Kernel, see [Embedding Generation Documentation](https://learn.microsoft.com/en-us/semantic-kernel/concepts/vector-store-connectors/embedding-generation?pivots=programming-language-csharp).
+
+**Upserting Records** - The demo uses the connector's `UpsertAsync()` method to insert or update records in the collection:
+```csharp
+await collection.UpsertAsync(glossaryEntries);
+```
+
+This creates 6 sample glossary entries with technical terms, generates embeddings for each definition, and stores them in Couchbase with the following structure:
+
+**Document ID:** `"1"` (from Key field)
+**Document Content:**
+```json
+{
+  "Category": "Software",
+  "Term": "API",
+  "Definition": "Application Programming Interface. A set of rules...",
+  "DefinitionEmbedding": [0.123, -0.456, 0.789, ...] // 1536 floats
+}
+```
+
+### Step 3: Hyperscale Index Creation
+
+This demo uses a **Hyperscale Vector Index** - optimized for pure vector searches without heavy scalar filtering. After documents are inserted, the demo creates the Hyperscale index:
+
+```sql
+CREATE VECTOR INDEX `hyperscale_glossary_index` 
+ON `demo`.`semantic-kernel`.`glossary` (DefinitionEmbedding VECTOR) 
+INCLUDE (Category, Term, Definition)
+USING GSI WITH {
+    "dimension": 1536,
+    "similarity": "cosine", 
+    "description": "IVF,SQ8"
+}
+```
+
+**Hyperscale Index Configuration:**
+- **Index Type**: Hyperscale Vector Index - best for pure vector similarity searches
+- **Vector Field**: `DefinitionEmbedding` (1536 dimensions)
+- **Similarity**: `cosine` (optimal for OpenAI embeddings)
+- **Include Fields**: Non-vector fields for faster retrieval
+- **Quantization**: `IVF,SQ8` (Inverted File with 8-bit scalar quantization)
+
+> **Note**: [Composite vector indexes](https://docs.couchbase.com/server/current/vector-index/composite-vector-index.html) can be created similarly by adding scalar fields to the index definition. Use Composite indexes when your queries frequently filter on scalar values before vector comparison. For this demo, we use Hyperscale since we're demonstrating pure semantic search capabilities.
+
+### Step 4: Vector Search Operations
+
+The demo performs two types of searches using the connector's `SearchAsync()` method with the Hyperscale index:
+
+#### Pure Vector Search
+
+Using the connector's search API:
+```csharp
+// Generate embedding from search query
+var searchVector = (await embeddingGenerator.GenerateAsync(
+    "What is an Application Programming Interface?")).Vector;
+
+// Search using the connector
+var results = await collection.SearchAsync(searchVector, top: 1)
+    .ToListAsync();
+```
+
+Behind the scenes, this executes a SQL++ query with `ANN_DISTANCE`:
+```sql
+SELECT META().id AS _id, Category, Term, Definition,
+       ANN_DISTANCE(DefinitionEmbedding, [0.1,0.2,...], 'cosine') AS _distance
```

Comment:
Highlight that the distance metric comes from the SimilarityMetric defined in the class instantiation